### PR TITLE
BYOC: Add guide to prepare AWS role

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/giantswarm/issues/2652

### CODE REVIEW

**Comment by marians** [comment]
We found a mistake in our thinking, which is also manifested in this draft. We explained this so that the roles to be created have to be bound to account ID `084190472784` (which is Giant Swarm's account). Instead, the roles have to be bound to the account ID the installation's host cluster is running in.

This means, the draft has to be changed in order to explain one further step: Determine your AWS account ID used for the host cluster, then apply this ID in the role's assignment.

File: content/guides/prepare-aws-account-for-guest-clusters/index.md [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+role must have Giant Swarm's account as a trusted entity, instead of the account
+running the host cluster, and it can have multi-factor authentication enabled.
+
+### 1. Basic role setup
+
+- Like above, go to the [Roles](https://console.aws.amazon.com/iam/home#/roles)
+subsection of the AWS console and select **Create role**. When asked to
+**Select type of trusted entity** chose **Another AWS account**.
+
+- In **Account ID** enter the value `084190472784`.
```
Comment (TheoBrigitte) [inline content/guides/prepare-aws-account-for-guest-clusters/index.md]: `084190472784` is no longer valid, as we want to use the host cluster account where aws-operator runs.

Comment (marians) [reply content/guides/prepare-aws-account-for-guest-clusters/index.md]: This here is the `GiantSwarmAdmin` role. This is for our team. And our team belongs to the 084190472784 account.

You are right with regard to the role aws-operator assumes, which is explained further up (from line 89 currently).

**Comment by marians** [comment]
@giantswarm/team-magic To make this article publishable and independent of the API implementation, I changed the last section regarding configuration of the organization. Needs an approval in order to merge.

File: content/guides/prepare-aws-account-for-guest-clusters/index.md [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+tags = ["tutorial"]
++++
+
+# Prepare an AWS account to run Giant Swarm guest clusters
+
+As detailed in the [Architecture](/reference/giantswarm-aws-architecture/) docs,
+the guest clusters (the clusters running your Kubernetes workloads) in a Giant
+Swarm installation are running in an AWS account separate from the host cluster.
+This allows for great flexibility depending on the requirements and the usage
+scenario. For example, it allows that the host cluster is run in an AWS account
```
Comment (r7vme) [inline content/guides/prepare-aws-account-for-guest-clusters/index.md]: `it allows the host cluster to be running in an AWS account`

**Comment by marians** [comment]
Thanks @r7vme ! Updated the wording.

File: content/guides/prepare-aws-account-for-guest-clusters/index.md [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+AWS account that's going to run the Giant Swarm guest clusters.
+
+Please hand the identifiers (ARNs) of those roles to Giant Swarm or to your
+contact person working with Giant Swarm.
+
+Your organization in Giant Swarm will then be set up to use these roles.
+
+Note that we plan to provide API support for this in the near future. So
+if you have access to the API of the Giant Swarm installation, you will be
+able to directly configure an organization with these roles.
```
Comment (TheoBrigitte) [inline content/guides/prepare-aws-account-for-guest-clusters/index.md]: This is not true anymore as we have this functionality currently supported in the API.

Comment (marians) [reply content/guides/prepare-aws-account-for-guest-clusters/index.md]: Yeah, correct. I'll update soon.

Comment (marians) [reply content/guides/prepare-aws-account-for-guest-clusters/index.md]: done

**Comment by marians** [comment]
So, with the last section regarding API usage updated, does this get a full approval?

**Comment by marians** [comment]
Cool, thanks!

**Pull request merged**